### PR TITLE
feat: allow to specify a custom camera id

### LINKED ISSUE #269: Camera selection
If someone has multiple rear cameras, how would the user select which one he wants to use?

### PR DESCRIPTION
**Pull request by robertsLando**
Fixes #269

### CODE REVIEW

File: src/misc/camera.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ const narrowDownFacingMode = async camera => {
       case "front":
         return { deviceId: { exact: frontCamera.deviceId } };
       default:
-        return undefined;
+        return devices.find(d => d.deviceId === camera) ? { deviceId: { exact: camera } } : undefined
```
Comment (robertsLando) [inline src/misc/camera.js:55]:
```suggestion
        return devices.find(d => d.deviceId === camera) ? { deviceId: { exact: camera } } : undefined;
```

**Comment by robertsLando** [comment]
@gruhn ping?

**Comment by gruhn** [comment]
@robertsLando thank you for your contribution. This is probably the most requested feature but I still would prefer not to include it. Please read my comments on this in #269.

**Comment by robertsLando** [comment]
@gruhn I have read that but I think that even with the most accurate heuristic algorithm to choose the right camera you will always have some cases where the user cannot select the desired camera (and so you will need another PR to improve it 😕 ). IMO giving the demand of such feature and considering it's not breaking anything I really think you should consider this PR. 

Otherwise you will always see users asking for it in issues and it's a pity as this is such a good package

**Comment by gruhn** [comment]
> even with the most accurate heuristic algorithm to choose the right camera you will always have some cases where the user cannot select the desired camera

That's true. We can also not guarantee that the package is bug free. We can only wait for developers to report bugs and improve the code over time. Similarly the heuristic can never be perfect. We can only wait for developers to report issues and improve it over time.

> you will always see users asking for it in issues

That's annoying for everyone, but I'm counting on it.

> IMO giving the demand of such feature and considering it's not breaking anything I really think you should consider this PR.

I know it's not breaking anything and it's a trivial change. But as soon as this feature exists, developers will stop reporting issues. And that puts an end to the above mentioned project. 

That doesn't mean I'm not considering your PR. I strongly prefer my solution but it's also true that I'm too busy all the time. Maybe the easy fix is better for the community.

**Comment by robertsLando** [comment]
@gruhn I agree with all your statement above, I understand but giving your last one:

>  I'm too busy all the time

That's something that I undestand too working on a lot of OSS projects, that's why I prefer solutions that simply allows users to do what they ask, less issues, less stress :) 

Feel free to edit my PR if needed and thanks for your work on this package 🙏🏼

**Comment by robertsLando** [comment]
Updates on this?

**Comment by AuspeXeu** [comment]
> I know it's not breaking anything and it's a trivial change. But as soon as this feature exists, developers will stop reporting issues. And that puts an end to the above mentioned project.

Just to give you an example: for me, this policy of waiting for everyone to create an issue for each single corner case, I simply stopped using your package and hence won't be reporting issues for you either. Meaning, your policy to push devs to report issues might actually drive them away similarly to what your argue this PR would be doing. I hope you can reconsider your stance on this @gruhn

**Comment by lexlemmens** [comment]
Any updates? Actually facing same issue right now where the wrong camera is used on Android phone. Currently running fork with this feature.

**Comment by robertsLando** [comment]
@gruhn Kindly ping

**Comment by gruhn** [comment]
I gave in. Passing a deviceId is now possible via the [`constraints`](https://master--vue-qrcode-reader.netlify.app/api/QrcodeStream.html#constraints) prop.

**Pull request closed without merging**